File: vol2/ch19/LeetCode77-2.cpp
class Solution {
    vector<vector<int>> ans;
public:
    vector<vector<int>> combine(int n, int k) {
        vector<int> x;
        dfs(n,k,x,1);
        return ans;
    }
    void dfs(int n,int k,vector<int>&x,int j)  {			//ªÿÀ›À„∑®
        if(x.size()==k)
            ans.push_back(x);
        else {
            for(int j1=j;j1<=n;j1++) {
                x.push_back(j1);
                dfs(n,k,x,j1+1);
                x.pop_back();
            }
        }
    }
};

File: vol2/ch19/LeetCode77-2.py
class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        self.ans,x=[],[]
        self.dfs(n,k,x,1)
        return self.ans
    def dfs(self,n,k,x,i):							#回溯算法
        if len(x)==k:
            self.ans.append(copy.deepcopy(x))
        else:
            for j in range(i,n+1):
                x.append(j)
                self.dfs(n,k,x,j+1)
                x.pop()
